backend/src/main/kotlin/org/loculus/backend/service/submission/SubmissionDatabaseService.kt

```diff
@@ -1152,21 +1169,32 @@ private fun Transaction.findNewPreprocessingPipelineVersion(): Long? {
                 select max(pipeline_version) as version
                 from
                     ( -- Newer pipeline versions...
-                        select distinct pipeline_version
-                        from sequence_entries_preprocessed_data
-                        where pipeline_version > (select version from current_processing_pipeline)
+                        select distinct sep.pipeline_version
+                        from sequence_entries_preprocessed_data sep
+                        join sequence_entries se
+                            on se.accession = sep.accession
+                            and se.version = sep.version
+                        where 
+                            se.organism = ?
+                            and sep.pipeline_version > (select version from current_processing_pipeline
+                                                        where organism = ?)
                     ) as newer
                 where
                     not exists( -- ...for which no sequence exists...
                         select
                         from
                             ( -- ...that was processed successfully with the current version...
-                                select accession, version
-                                from sequence_entries_preprocessed_data
+                                select sep.accession, sep.version
+                                from sequence_entries_preprocessed_data sep
+                                join sequence_entries se
```

Comment:
To avoid this join, we might as well add the organism to the processed table, it's just one column - a little bit of denormalization might not harm, especially as organism isn't supposed to change

Reply:
True, it would be good to make this query simpler, but now that this already works I'm not sure I would make this change to this table (and I won't unless you insist). In other places, other potential changes to the handling of organisms have been mentioned, and so I would rather not denormalize so we can avoid renormalizing in the future, which would just be wasted effort on both ends.

Reply:
(I like normalization personally [even here])